LEGO1/mxsmack.cpp

```diff
@@ -140,15 +139,88 @@ void MxSmack::Destroy(MxSmack* p_mxSmack)
 		delete[] p_mxSmack->m_unk0x6b4;
 }
 
-// STUB: LEGO1 0x100c5db0
-void MxSmack::FUN_100c5db0(
+// This should be refactored to somewhere else
+inline MxLong AbsFlipped(MxLong p_value)
+{
+	return p_value > 0 ? p_value : -p_value;
+}
+
+// FUNCTION: LEGO1 0x100c5db0
+MxResult MxSmack::LoadFrame(
 	MxBITMAPINFO* p_bitmapInfo,
 	MxU8* p_bitmapData,
 	MxSmack* p_mxSmack,
 	MxU8* p_chunkData,
-	MxBool p_und,
+	MxBool p_paletteChanged,
 	MxRectList* p_list
 )
+{
+	p_bitmapInfo->m_bmiHeader.biHeight = -AbsFlipped(p_bitmapInfo->m_bmiHeader.biHeight);
+	*(MxU8**) (p_mxSmack->m_unk0x6b4 + 4) = p_bitmapData;
+
+	// Reference: https://wiki.multimedia.cx/index.php/Smacker#Palette_Chunk
+	if (p_paletteChanged) {
+		MxU8 palette[772];
+
+		MxU8* intoChunk = p_chunkData + 1;
+		MxU8* intoPalette = palette;
+		MxU16 paletteIndex = 0;
+		// TODO: struct incorrect, Palette at wrong offset?
+		MxU8* currentPalette = &p_mxSmack->m_smackTag.Palette[4];
+
+		do {
+			if (*intoChunk & 0x80) {
```

Comment:
This may help:
https://github.com/FFmpeg/FFmpeg/blob/45b4781e9a19cb4c9b2bf6e7018087698067fb74/libavformat/smacker.c#L268C7-L276C42